src/server/list_family.cc

```diff
@@ -565,6 +563,66 @@ void ListFamily::LLen(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ListFamily::LPos(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 1);
+  string_view elem = ArgS(args, 2);
+
+  int rank = 1;
+  uint32_t count = 1;
+  uint32_t max_len = 0;
+  bool skip_count = true;
+
+  for (size_t i = 3; i < args.size(); i++) {
+    ToUpper(&args[i]);
+    const auto& arg_v = ArgS(args, i);
+    if (arg_v == "RANK") {
+      if (!absl::SimpleAtoi(ArgS(args, (i + 1)), &rank) || rank == 0) {
+        return (*cntx)->SendError(kInvalidIntErr);
+      }
+    }
+    if (arg_v == "COUNT") {
+      if (!absl::SimpleAtoi(ArgS(args, (i + 1)), &count)) {
+        return (*cntx)->SendError(kInvalidIntErr);
+      }
+      skip_count = false;
+    }
+    if (arg_v == "MAXLEN") {
+      if (!absl::SimpleAtoi(ArgS(args, (i + 1)), &max_len)) {
+        return (*cntx)->SendError(kInvalidIntErr);
+      }
+    }
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpPos(t->GetOpArgs(shard), key, elem, rank, count, max_len);
+  };
+
+  Transaction* trans = cntx->transaction;
+  OpResult<vector<uint32_t>> result = trans->ScheduleSingleHopT(std::move(cb));
+
+  if (result.status() == OpStatus::WRONG_TYPE) {
+    return (*cntx)->SendError(result.status());
+  } else if (result.status() == OpStatus::INVALID_VALUE) {
+    return (*cntx)->SendError(result.status());
+  }
+
+  switch (result->size()) {
+    case 0:
```

Comment:
this is still incorrect. you should only use "SendNull when  empty under `skip_count` branch.
the "else" code works well for empty case as is.

Reply:
```cpp
if (skip_count) {
  if (result->empty()) {
    (*cntx)->SendNull();
  } else {
    (*cntx)->SendLong((*result)[0]);
  }
} else {
   (*cntx)->StartArray(result->size());
    const auto& array = result.value();
    for (const auto& v : array) {
      (*cntx)->SendLong(v);
    }
}
```